xonsh/prompt/gitstatus.py

```diff
@@ -136,25 +120,32 @@ def _gitoperation(gitdir):
         ("REVERT_HEAD", "REVERTING"),
         ("BISECT_LOG", "BISECTING"),
     )
-    return [f[1] for f in files if os.path.exists(os.path.join(gitdir, f[0]))]
+    operations = [f[1] for f in files if os.path.exists(os.path.join(gitdir, f[0]))]
+    if operations:
+        return "|" + "|".join(operations)
+    return ""
+
 
+def _get_operation_stashed():
+    gitdir = _check_output(["git", "rev-parse", "--git-dir"]).strip()
+    stashed = _get_stash(gitdir)
+    operation = _gitoperation(gitdir)
+    return {"operation": operation, "stashed": stashed}
+
+
+def _get_status_fields():
+    """Return parsed values from ``git status``"""
 
-def gitstatus():
-    """Return namedtuple with fields:
-    branch name, number of ahead commit, number of behind commit,
-    untracked number, changed number, deleted number, conflicts number,
-    staged number, stashed number, operation."""
     status = _check_output(["git", "status", "--porcelain", "--branch"])
     branch = ""
-    num_ahead, num_behind = 0, 0
+    ahead, behind = 0, 0
     untracked, changed, deleted, conflicts, staged = 0, 0, 0, 0, 0
-    diffchanged = _get_files_changed()
     for line in status.splitlines():
         if line.startswith("##"):
             line = line[2:].strip()
             if "Initial commit on" in line:
                 branch = line.split()[-1]
-            elif "no branch" in line:
+            elif "no branch" in line and not _is_hidden("hash"):
```

Comment:
The 'tag or hash' should still be used regardless of `_is_hidden("hash")`.
The HASH symbol in DEFS actually means the symbol to say we're showing a hash instead of a branch.
For instance when `HASH=":"` and we're in a detached head:
![image](https://user-images.githubusercontent.com/18242949/120085229-fc1d8d80-c0de-11eb-853c-3da4c60a8e62.png)

What I meant was renaming the HASH symbol in DEFS to `HASH_INDICATOR` and check for `_is_hidden("hash_indicator")` inside `_get_tag_or_hash` where it's used.
So that if `$XONSH_GITSTATUS_FIELDS_TO_HIDE = ['HASH_INDICATOR']` the indicator won't be used to indicate a hash.

Reply:
I think we can remove the `hash` from the hide-fields setting. Since it is covered by `branch`